### PR TITLE
[SPARK-42712][PYTHON][DOC] Improve docstring of mapInPandas and mapInArrow

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?
Improve docstring of mapInPandas and mapInArrow 

### Why are the changes needed?
For readability. We call out they are not scalar - the input and output of the function might be of different sizes.

### Does this PR introduce _any_ user-facing change?
No. Doc change only.

### How was this patch tested?
Existing tests.

### CODE REVIEW

**Comment by xinrong-meng** [comment]
CC @HyukjinKwon @hvanhovell

**Review by zhengruifeng — APPROVED** [review]

File: python/pyspark/sql/pandas/map_ops.py [hunk 1]
```diff
@@ -44,7 +44,8 @@ def mapInPandas(
         together as an iterator of `pandas.DataFrame`\\s to the function and the
         returned iterator of `pandas.DataFrame`\\s are combined as a :class:`DataFrame`.
         Each `pandas.DataFrame` size can be controlled by
-        `spark.sql.execution.arrow.maxRecordsPerBatch`.
+        `spark.sql.execution.arrow.maxRecordsPerBatch`. The size of the function's input and
+        output might be different.
```
Comment (HyukjinKwon) [inline python/pyspark/sql/pandas/map_ops.py:48]:
```suggestion
        output can be different.
```

File: python/pyspark/sql/pandas/map_ops.py [hunk 2]
```diff
@@ -108,7 +109,8 @@ def mapInArrow(
         together as an iterator of `pyarrow.RecordBatch`\\s to the function and the
         returned iterator of `pyarrow.RecordBatch`\\s are combined as a :class:`DataFrame`.
         Each `pyarrow.RecordBatch` size can be controlled by
-        `spark.sql.execution.arrow.maxRecordsPerBatch`.
+        `spark.sql.execution.arrow.maxRecordsPerBatch`. The size of the function's input and
+        output might be different.
```
Comment (HyukjinKwon) [inline python/pyspark/sql/pandas/map_ops.py:113]:
```suggestion
        output can be different.
```

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.4.

**Pull request closed without merging**